Product/Production/Services/DirectCore/src/main/java/gov/hhs/fha/nhinc/direct/MimeMessageBuilder.java

```diff
@@ -233,202 +220,17 @@ public MimeMessage build() {
-	}
-	
-	/**
-	 * Get a FileOutputStream for a specified file.
-	 * 
-	 * @param xmlFile
-	 * 		Contains the file for which to get a FileOutputStream.
-	 * @return
-	 * 		Returns a FileOutputStream object.
-	 * @throws FileNotFoundException
-	 */
-	protected FileOutputStream getXmlAttachmentFileOutputStream(File xmlFile) throws FileNotFoundException {
-		return new FileOutputStream(xmlFile);
-	}
-
-	/**
-	 * Get the XML attachment file name.
-	 * 
-	 * @param fileName
-	 * 		Contains the name of the file.
-	 * @return
-	 * 		Returns a File object.
-	 */
-	protected File getXmlAttachmentFile(String fileName) {
-		return new File(fileName);
-	}
 	
-    /**
-     * Format the "messageId" to be used to name the XDM attachment.
-     * 
-     * @param messageId
-     * 		Contains a "messageId" to format.
-     * @return
-     * 		Returns a formatted "messageId".
-     */
-    private String formatMessageIdForXDMAttachmentName(String messageId) {
-    	String formattedMessageId = messageId;
-    	
-    	LOG.debug("MimeMessageBuilder.formatMessageIdForXDMAttachmentName - Passed in value: '" + messageId + "'");
-    	
-    	if (NullChecker.isNotNullish(messageId)) {
-    		formattedMessageId = messageId.replace(NhincConstants.WS_SOAP_HEADER_MESSAGE_ID_PREFIX, "");	
-    		
-    		if (formattedMessageId.startsWith("<")) {
-    			formattedMessageId = formattedMessageId.substring(1);
-			}
-    		
-    		if (formattedMessageId.endsWith(">")) {
-    			formattedMessageId = formattedMessageId.substring(0, formattedMessageId.length() - 1);
-			}
-		}
-    	
-    	LOG.debug("MimeMessageBuilder.formatMessageIdForXDMAttachmentName - Return value: '" + formattedMessageId + "'");
-     	
-		return formattedMessageId;
-	}
-
-	/**
-     * Get the mime type suffix.
-     * 
-     * @param mimeType
-     * 		Contains the mime type for which to get the suffix.
-     * @return
-     * 		Returns a String of the mime suffix.
-     */
-    private String getSuffix(String mimeType) {
-        return "." + MimeType.lookup(mimeType).getSuffix();
-    }
-    
-    
 	/**
-	 * Get the direct attachment option from a properties file.
+	 * Get a handle to a AttachmentHandler object.
 	 * 
 	 * @return
-	 * 		Returns the direct attachment option.
+	 * 		Returns a handle to a AttachmentHandler object.
 	 */
-	protected String getDirectAttachmentOption() {
-		String directAttachmentOption = "";
-		
-		try {
-			directAttachmentOption = PropertyAccessor.getInstance(GATEWAY_PROPERTIES_FILE).getProperty(DIRECT_ATTACHMENT_OPTION);
-		} catch (PropertyAccessException e) {
-			e.printStackTrace();
-			LOG.error("Error occured in retrieving the direct attachment option. Defaulting to 'xdm'.");
-			
-			directAttachmentOption = "xdm";
-		}
-		
-		LOG.debug("Direct: Direct attachment option is: '" + directAttachmentOption + "'");	
-		
-		return directAttachmentOption;
+	protected AttachmentHandler getAttachmentHandler() {
+		return new AttachmentHandler();
```

Comment:
In other Aurion code - I have seen this be a property of the class and then in the getter there is an if null block to return new.  This one will create a new instance each time instead of reusing an existing instance.  Is that intentional?